pkg/genericapiserver/destinations.go

```diff
@@ -0,0 +1,189 @@
+
+	// EtcdLocationOverrides contains the list of "special" locations that are used for particular GroupResources
+	// these are merged on top of the default DefaultEtcdConfig when requesting the storage.Interface for a given GroupResource
+	EtcdLocationOverrides map[unversioned.GroupResource][]string
+
+	// EtcdPrefixOverrides contains the list of "special" prefixes for a GroupResource.  Resource=* means for the entire group
+	EtcdPrefixOverrides map[unversioned.GroupResource]string
+
+	// DefaultSerializer is used to create encoders and decoders for the storage.Interface.
+	DefaultSerializer runtime.NegotiatedSerializer
+
+	// SerializerOverrides contains the list of "special" serializers for a GroupResource.  Resource=* means for the entire group
+	SerializerOverrides map[unversioned.GroupResource]runtime.NegotiatedSerializer
+
+	// ResourceEncodingConfig describes how to encode a particular GroupVersionResource
+	ResourceEncodingConfig ResourceEncodingConfig
+
+	// CohabitatingResources keeps track of which resources must be stored together.  This happens when we have multiple ways
+	// of exposing one set of concepts.  autoscaling.HPA and extensions.HPA as a for instance
+	// the order of the slice matters!  its the priority order of lookup for finding a storage location
+	CohabitatingResources map[unversioned.GroupResource][]unversioned.GroupResource
+
+	// APIResourceConfigSource indicate whether the *storage* is enabled, NOT the API
+	APIResourceConfigSource APIResourceConfigSource
+}
+
+const AllResources = "*"
+
+func NewStorageDestinations(defaultEtcdConfig etcdstorage.EtcdConfig, defaultSerializer runtime.NegotiatedSerializer, resourceEncodingConfig ResourceEncodingConfig, resourceEnablement APIResourceConfigSource) StorageDestinations {
+	return StorageDestinations{
+		DefaultEtcdConfig:       defaultEtcdConfig,
+		EtcdLocationOverrides:   map[unversioned.GroupResource][]string{},
+		EtcdPrefixOverrides:     map[unversioned.GroupResource]string{},
+		DefaultSerializer:       defaultSerializer,
+		SerializerOverrides:     map[unversioned.GroupResource]runtime.NegotiatedSerializer{},
+		ResourceEncodingConfig:  resourceEncodingConfig,
+		CohabitatingResources:   map[unversioned.GroupResource][]unversioned.GroupResource{},
+		APIResourceConfigSource: resourceEnablement,
+	}
+}
+
+func (s *StorageDestinations) SetEtcdLocation(groupResource unversioned.GroupResource, location []string) {
+	s.EtcdLocationOverrides[groupResource] = location
+}
+
+func (s *StorageDestinations) SetEtcdPrefix(groupResource unversioned.GroupResource, prefix string) {
+	s.EtcdPrefixOverrides[groupResource] = prefix
+}
+
+func (s *StorageDestinations) SetSerializer(groupResource unversioned.GroupResource, serializer runtime.NegotiatedSerializer) {
+	s.SerializerOverrides[groupResource] = serializer
+}
+
+// AddCohabitatingResources links resources together the order of the slice matters!  its the priority order of lookup for finding a storage location
```

Comment:
If I understand correctly, I think it is expressing what was hard coded for jobs/hpas before… that jobs/batch and jobs/extensions should be stored together. I think that config belongs here, not at the rest handler edges, but could use explanation or example.